src/twister2/device/device_abstract.py

```diff
@@ -28,23 +35,40 @@ def env(self) -> dict[str, str]:
         return env
 
     @abc.abstractmethod
-    def connect(self):
+    def connect(self) -> None:
         pass
 
     @abc.abstractmethod
-    def disconnect(self):
+    def disconnect(self) -> None:
         pass
 
-    @abc.abstractmethod
     def flash(self, build_dir: str | Path, timeout: float = 60.0) -> None:
         """
         Flash and run code on a device.
 
         :param build_dir: build directory
         :param timeout: time out in seconds
         """
+        logger.info('Flashing device')
+        self.build_dir = build_dir
+        self.timeout = timeout
+        self.start()
 
     @property
     @abc.abstractmethod
     def out(self) -> Generator[str, None, None]:
         """Return output from a device."""
+
+    def join(self, timeout: float = None) -> None:
+        super().join(timeout=1)
+        if not self._loop.is_closed:
```

Comment:
Missing verification if `self._loop` is None - otherwise program crash during testing on hardware (where `self._loop` is not initialized).
```python
if (self._loop is not None) and (not self._loop.is_closed):
    self._loop.close()
```

Reply:
This code will be removed.